src/app/pages/MarginTradePage/components/NotificationSettingsDialog/index.tsx

```diff
@@ -76,6 +91,33 @@ export const NotificationSettingsDialog: React.FC<INotificationSettingsDialogPro
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [notificationUser]);
 
+  const updateSubscriptions = useCallback(() => {
+    if (notificationUser) {
+      const subscriptions: Subscription[] = JSON.parse(
+        JSON.stringify(notificationUser?.subscriptions),
+      );
+      const d1Notifications = Object.values(Notification);
+      d1Notifications.forEach(item => {
+        const foundSubscription = subscriptions.findIndex(
+          sub => sub.notification === item,
+        );
+        if (foundSubscription) {
```

Comment:
```suggestion
        if (foundSubscription >= 0) {
```
Currently crashes if you try to enable notifications with a new wallet as `foundSubscription === -1`, so next line fails